### PR TITLE
Make TableUpdateValidator composable in a sequence of operations.

### LINKED ISSUE #4683: Make TableUpdateValidator composable in a sequence of operations
As a developer of table operations, I want to insert TableUpdateValidators into a sequence of operations so that I can make sure that a failing or incorrect operation has valid input.

### PR DESCRIPTION
**Pull request by cpwright**
Closes #4683.

### CODE REVIEW

File: engine/table/src/main/java/io/deephaven/engine/table/impl/TableUpdateValidator.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import io.deephaven.util.SafeCloseableList;
 import org.apache.commons.lang3.mutable.MutableInt;
 
+import javax.sound.midi.Track;
```
Comment (nbauernfeind) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/TableUpdateValidator.java:22]: unused import

File: engine/table/src/main/java/io/deephaven/engine/table/impl/TableUpdateValidator.java [hunk 2]
```diff
@@ -97,11 +102,10 @@ public String getLogPrefix() {
 
     @Override
     public Result initialize(boolean usePrev, long beforeClock) {
-        rowSet = usePrev ? tableToValidate.getRowSet().copyPrev() : tableToValidate.getRowSet().copy();
+        rowSet = (usePrev ? tableToValidate.getRowSet().copyPrev() : tableToValidate.getRowSet().copy()).toTracking();
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/TableUpdateValidator.java:105]:
```suggestion
        rowSet = (usePrev ? tableToValidate.getRowSet().prev() : tableToValidate.getRowSet()).copy().toTracking();
```

File: engine/table/src/main/java/io/deephaven/engine/table/impl/TableUpdateValidator.java [hunk 3]
```diff
@@ -212,7 +216,13 @@ private void onUpdate(final TableUpdate upstream) {
                 }
                 result.append("\n");
                 resultTable.notifyListenersOnError(new RuntimeException(result.toString()), null);
+                return;
             }
+
+            final TableUpdateImpl downstream = TableUpdateImpl.copy(upstream);
+            downstream.modifiedColumnSet = resultTable.getModifiedColumnSetForUpdates();
+            transformer.transform(upstream.modifiedColumnSet(), downstream.modifiedColumnSet);
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/TableUpdateValidator.java:224]: I believe this to be unnecessary. The source table cannot safely mutate its reusable MCS until the next cycle.

File: engine/table/src/main/java/io/deephaven/engine/table/impl/TableUpdateValidator.java [hunk 4]
```diff
@@ -97,11 +102,10 @@ public String getLogPrefix() {
 
     @Override
     public Result initialize(boolean usePrev, long beforeClock) {
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/TableUpdateValidator.java:104]:
```suggestion
    public Result<QueryTable> initialize(boolean usePrev, long beforeClock) {
```

File: engine/table/src/test/java/io/deephaven/engine/table/impl/ErrorListener.java [hunk 5]
```diff
@@ -16,9 +16,7 @@ public ErrorListener(Table table) {
     }
 
     @Override
-    public void onUpdate(final TableUpdate upstream) {
-        TestCase.fail("Should not have gotten an update!");
-    }
+    public void onUpdate(final TableUpdate upstream) {}
```
Comment (rcaudy) [inline engine/table/src/test/java/io/deephaven/engine/table/impl/ErrorListener.java:19]: You have created an unused import.

File: engine/test-utils/src/main/java/io/deephaven/engine/testutil/EvalNugget.java [hunk 6]
```diff
@@ -88,10 +96,11 @@ public void onFailureInternal(Throwable originalException, Entry sourceEntry) {
     }
 
     private final TableUpdateValidator validator;
+    private final TableUpdateListener validationFailureListener = new ValidationFailureListener();
```
Comment (rcaudy) [inline engine/test-utils/src/main/java/io/deephaven/engine/testutil/EvalNugget.java:99]: Should we bother constructing this if we won't construct a validator?

**Review by rcaudy — APPROVED** [review]

**Pull request merged**